### DISCUSSION TITLE
The Executable made by pyinstaller is not working!

### DISCUSSION THREAD

**Discussion by OverclockedSenku** [pyinstaller/pyinstaller#8409] [Help]
Hi, I am using termux and have pyinstaller installed I have built my py script with `pyinstaller -F *.py -n Codnium` I am building this in the dir `/data/data/com.termux/files/home/Projects/PyBox/` this is the project dir where I am executing the build command. After a successful build the project directory has `dist/Codnium` when I run the executable `/data/data/com.termux/files/home/Projects/PyBox/dist/Codnium` it runs but when I move the executable to another directory for example: `/data/data/com.termux/files/home/.local/bin/Codnium` and run it, it gives me this error 
```
/data/data/com.termux/files/home/.local/bin/Codnium: strtab out of bounds error; STRSZ=0, name=608
zsh: abort      Codnium
```

This program Is interactive program so it does not access the `sys.argv[1]`

**Comment by bwoodsend** [comment]
This doesn't really tell us anything we can work with. If moving it breaks it then you're code is presumably not relocatable which usually means that you're referencing data files invalidly (relative paths instead of [using `__file__`](https://pyinstaller.org/en/stable/runtime-information.html#using-file)). Beyond that, you'll have to do your own debugging, even if it's just adding print statements or commenting out lines, to figure out where the error originates from.

**Reply by OverclockedSenku** [reply]
My script is an program that uses requests to communicate between my website backend api. It's like chatting with server and it doesn't read or write to user's local storage or folder

**Reply by rokm** [reply]
Do you get the same error if you freeze a simple `print("hello world")` program, and move it?

**Reply by OverclockedSenku** [reply]
Yes, a simple hello World works. But my program doesn't.
Here is my code I am trying to run as binary


```python
import Console
import requests

url = 'http://localhost:8000/api/chat'

headers = {
    # headers are here that are in String, String format.
}

def main():
    start()

def start():
    Console.info("🚀 Starting Codnium")
    Console.success("🚀 Codnium Started Successfully")
    Console.info("If you want to end chat just give prompt \"bye\"")
    chatting = True
    while chatting:
        unput = Console.getUserInput("You: ")
        chat(unput)
        if(unput.__contains__("bye")):
            chatting = False



def chat(prompt: str):
    payload = {
        "messages": [
            {
                "id": "",
                "content": prompt,
                "role": "user"
            }
        ],
        "id": "testing",
        "userId": "testing"
        }

    response = requests.post(url, headers=headers, json=payload)

    if response.status_code == 200:
        content_encoding = response.headers.get('Content-Encoding', 'identity')

        if content_encoding == 'br':
            Console.success("Codnium: ")
            Console.out(response.content.decode("utf-8"))
        else:
            Console.success("Codnium: ")
            Console.out(response.text)
    else:
        Console.error(f'Request failed with status code {response.status_code}')


if __name__ == "__main__":
    main()


#decompressed_data = brotli.decompress(response.content)
        #res = decompressed_data.decode('utf-8')
        #print(res)               


```                                                      

Here the code of console:
```python
RED = "\033[91m"
GREEN = "\033[92m"
YELLOW = "\033[93m"
BLUE = "\033[94m"
RESET = "\033[0m"

def info(toOut: str):
    print("%s %s %s" % (BLUE, toOut, RESET))

def error(toOut: str):
    print("%s %s %s" % (RED, toOut, RESET))

def success(toOut: str):
    print("%s %s %s" % (GREEN, toOut, RESET))

def warn(toOut: str):
    print("%s %s %s" % (YELLOW, toOut, RESET))

def out(toOut: str):
    print(toOut)

def getUserInput(promt: str = ""):
    prompt = promt
    if prompt == "":
        prompt = " "
    else:
        prompt = " " + prompt

    return input(prompt)
```

**Reply by bwoodsend** [reply]
That code as-is shows the error? Because the most likely candidate in there for some kind of C out of bounds error is brotli which you've disabled in your example.

**Reply by rokm** [reply]
> `/data/data/com.termux/files/home/.local/bin/Codnium: strtab out of bounds error; STRSZ=0, name=608`

Hmm, the `strtab out of bounds error; STRSZ` seems to be coming from android's linker: https://android.googlesource.com/platform/bionic.git/+/master/linker/linker_soinfo.cpp#766

That `STRSZ=0` implies that `strtab_size_` was 0, meaning that it could not possibly contain any string...

--

Can you try rebuilding with `--debug bootloader` added to you PyInstaller command, so we can see if the crash happens in the parent or the child process, and if it is triggered by bootloader itself or from the python code (e.g., a shared library being loaded from python - although since the executable name is printed in the error message, the chances of that are somewhat slim).

**Comment by OverclockedSenku** [comment]
> That code as-is shows the error? Because the most likely candidate in there for some kind of C out of bounds error is brotli which you've disabled in your example.



> That code as-is shows the error? Because the most likely candidate in there for some kind of C out of bounds error is brotli which you've disabled in your example.

Yeah, I was first using brotli but I forgot that server sends decoded response when using testing client.

**Comment by OverclockedSenku** [comment]
It finally fixed, it was the problem with my primary phone, when I built it from another phone it worked and even I can now execute it with the android's shell environment without termux. In my primary phone when I run the binary with that debug options, the output is `Segmentation Fault`. Thank you guys and Sorry for my inconvenience.
